src/libraries/System.Linq.Expressions/src/System/Dynamic/Utils/Helpers.cs

```diff
@@ -3,12 +3,14 @@
 // See the LICENSE file in the project root for more information.
 
 using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
 
 namespace System.Dynamic.Utils
 {
     // Miscellaneous helpers that don't belong anywhere else
     internal static class Helpers
     {
+        [return: MaybeNull]
         internal static T CommonNode<T>(T first, T second, Func<T, T> parent) where T : class
```

Comment:
Since `T` is constrained to `class`, the method can return `T?` instead.

Reply:
Good point, thanks